### PR TITLE
Initial structs and constants

### PR DESCRIPTION
**Pull request by MikeCamel**
Signed-off-by: MikeCamel <mike@p2ptrust.org>

### CODE REVIEW

**Review by mbestavros — DISMISSED** [review]
Seems reasonable to me.

File: Cargo.toml [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+authors = ["Nathaniel McCallum <npmccallum@redhat.com>", "MikeCamel <mike@p2ptrust.org"]
 edition = "2018"
+license = "Apache-2.0"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+serde = "1.0"
+serde_cbor = "0.10"
+serde_derive = "1.0"
```
Comment (connorkuehl) [inline Cargo.toml]:
```suggestion
serde = { version = "1.0", features = ["derive"] }
serde_cbor = "0.10"
```

File: src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use serde_derive::{Deserialize, Serialize};
```
Comment (connorkuehl) [inline src/lib.rs]:
```suggestion
use serde::{Deserialize, Serialize};
```

File: src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,91 @@
...
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use uuid::Uuid;
+
+pub const LOCAL_LISTEN_ADDRESS: &str = "192.168.1.202";
```
Comment (connorkuehl) [inline src/lib.rs]: What about using the default address: `0.0.0.0`?

File: src/lib.rs [hunk 4]
```diff
@@ -1,3 +1,91 @@
...
+
+pub const PROTO_VERSION: f32 = 0.1;
+pub const PROTO_NAME: &str = "Enarx-Keep-Manager";
+pub const BIND_PORT: u16 = 3030;
+
+pub const KEEP_LOADER_STATE_UNDEF: u8 = 0;
+pub const KEEP_LOADER_STATE_LISTENING: u8 = 1;
+pub const KEEP_LOADER_STATE_STARTED: u8 = 2;
+pub const KEEP_LOADER_STATE_COMPLETE: u8 = 3;
+pub const KEEP_LOADER_STATE_ERROR: u8 = 15;
```
Comment (connorkuehl) [inline src/lib.rs]:
```suggestion
pub enum LoaderState {
    Indeterminate,
    Ready,
    Running,
    Shutdown,
    Error,
}
```

File: src/lib.rs [hunk 5]
```diff
@@ -1,3 +1,91 @@
...
+pub struct Workload {
+    pub wasm_binary: Vec<u8>,
+    pub human_readable_info: String,
+}
+
+//TODO - rename in favour of cbor, possibly remove
+#[derive(Serialize, Deserialize, Clone)]
+pub struct Command {
+    pub commandtype: String,
+    pub commandcontents: String,
```
Comment (connorkuehl) [inline src/lib.rs:81]:
```suggestion
    pub kind: String,
    pub contents: String,
```

File: src/lib.rs [hunk 6]
```diff
@@ -1,3 +1,91 @@
...
+pub const KEEP_PORT: &str = "keep-port";
+pub const KEEP_ADDR: &str = "keep-addr";
+pub const KEEP_KUUID: &str = "kuuid";
+pub const KEEP_ARCH: &str = "keep-arch";
+pub const WASMLDR_BIND_PORT_CMD: &str = "wasmldr-bind-port";
+pub const WASMLDR_ADDR_CMD: &str = "wasmldr-addr";
+pub const KEEP_ARCH_NIL: &str = "nil";
+pub const KEEP_ARCH_SEV: &str = "sev";
+pub const KEEP_ARCH_SGX: &str = "sgx";
+pub const KEEP_ARCH_KVM: &str = "kvm";
```
Comment (connorkuehl) [inline src/lib.rs]:
```suggestion
pub enum Backend {
    Nil,
    Sev,
    Sgx,
    Kvm,
}
```

**Review by lkatalin — DISMISSED** [review]

File: src/lib.rs [hunk 7]
```diff
@@ -1,3 +1,91 @@
...
+pub const KEEP_INFO_COMMAND: &str = "keep-info";
+pub const CONTRACT_COMMAND: &str = "command";
+pub const KEEP_COMMAND: &str = "command";
+pub const KEEP_AUTH: &str = "auth-token";
+pub const KEEP_PORT: &str = "keep-port";
+pub const KEEP_ADDR: &str = "keep-addr";
+pub const KEEP_KUUID: &str = "kuuid";
+pub const KEEP_ARCH: &str = "keep-arch";
+pub const WASMLDR_BIND_PORT_CMD: &str = "wasmldr-bind-port";
+pub const WASMLDR_ADDR_CMD: &str = "wasmldr-addr";
```
Comment (connorkuehl) [inline src/lib.rs:32]: Are these meant to be MIME-type packet IDs?

**Review by connorkuehl — CHANGES_REQUESTED** [review]
Do you have additional context for these changes? I understand these are supposed to be message buffers but additional information like communication flows, etc, would be helpful in forming additional feedback.

File: Cargo.toml (hunk 1 above)
Comment (MikeCamel) [reply Cargo.toml]: Taken.

File: src/lib.rs (hunk 2 above)
Comment (MikeCamel) [reply src/lib.rs]: Done.

File: src/lib.rs (hunk 3 above)
Comment (MikeCamel) [reply src/lib.rs]: Changed.

File: src/lib.rs (hunk 4 above)
Comment (MikeCamel) [reply src/lib.rs]: Many of the entries here are based on an initial PoC implementation.  I've been meaning to move to Enums: taking these.

File: src/lib.rs (hunk 5 above)
Comment (MikeCamel) [reply src/lib.rs:81]: I'm going to leave this (as it's working now with current code) until we've worked out how to do this properly.

File: src/lib.rs (hunk 6 above)
Comment (MikeCamel) [reply src/lib.rs]: Taken - see above.

File: src/lib.rs (hunk 7 above)
Comment (MikeCamel) [reply src/lib.rs:32]: No - they're what I've been using till now.  I'm going to keep them as is for now until we start transitioning.

**Review by connorkuehl — APPROVED** [review]

**Review by lkatalin — APPROVED** [review]

**Review by mbestavros — APPROVED** [review]

**Pull request merged**